wallet/src/main/java/com/android/identity_credential/wallet/PresentationActivity.kt

```diff
@@ -112,30 +111,42 @@ class PresentationActivity : FragmentActivity() {
             this.deviceEngagement = deviceEngagement
             Logger.i(TAG, "engagement info set")
 
-            launchPresentationActivity(context)
-            state.value = State.CONNECTED
+            if (phase.value != Phase.ENGAGING) {
+                launchPresentationActivity(context)
+            }
+            phase.value = Phase.CONNECTED
         }
 
         private fun launchPresentationActivity(context: Context) {
             val launchAppIntent = Intent(context, PresentationActivity::class.java)
             launchAppIntent.addFlags(
                 Intent.FLAG_ACTIVITY_NEW_TASK or
                         Intent.FLAG_ACTIVITY_NO_HISTORY or
-                        Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS
+                        Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS or
+                        Intent.FLAG_ACTIVITY_NO_ANIMATION
             )
             context.startActivity(launchAppIntent)
         }
 
         fun isPresentationActive(): Boolean {
-            return state.value != State.NOT_CONNECTED
+            return phase.value != Phase.NOT_CONNECTED
+        }
+
+        fun stopPresentationReaderTimeout(context: Context) {
```

Comment:
(see comment above for more details) -- I think this function should check for `phase.value != Phase.NOT_CONNECTED` and then proceed with the statements below

Reply:
It's quite intentional that `PresentationActivity.isPresentationActive()` exists... the current phase needs to be exposed to `NfcEngagementHandler` to avoid situations where the user is tapping but we're already in the middle of a presentation. So I don't think this should be changed.